src/listening_groups.py

```diff
@@ -0,0 +1,61 @@
+import csv
+import os.path
+
+from core_data_modules.logging import Logger
+from core_data_modules.traced_data import Metadata
+import time
+
+from src.lib import PipelineConfiguration
+
+log = Logger(__name__)
+
+class ListeningGroup(object):
+
+    @classmethod
+    def tag_listening_groups_participants(cls, user, data, listening_group_dir):
+        '''
+        This tags uids who participated in repeat listening groups and/or weekly listening
+        group sessions.
+        :param user: Identifier of the user running this program, for TracedData Metadata.
+        :type user: str
+        :param data: TracedData objects to tag listening group participation to.
+        :type data: iterable of TracedData
+        :param listening_group_dir: Directory containing de-identified listening groups contacts CSVs.
+        :type user: str
+        '''
+
+        repeat_listening_group_participants = [] # Contains uids of listening group participants who will participate
+                                                 # in all listening group sessions.
+        listening_group_participants = dict()   # Contains lists of weekly listening group participants. The participants
+                                                # will change each week.
+
+        with open(f'{listening_group_dir}/repeat_listening_group.csv', "r", encoding='utf-8-sig') as f:
+            repeat_listening_group_data = list(csv.DictReader(f))
+            for row in repeat_listening_group_data:
+                repeat_listening_group_participants.append(row['Mobile No'])
```

Comment:
Can this be an avf-phone-uuid rather than a mobile number?

Reply:
Yes, updated thanks!